This is a backport of http://www.linux-mips.org/g/linux/db423f6e (which
will be in 2.6.23) to 2.6.22:


[MIPS] Fix CONFIG_BUILD_ELF64 kernels with symbols in CKSEG0.

Ralf Baechle [Tue, 11 Sep 2007 07:50:40 +0000 (08:50 +0100)]

The __pa() for those did assume that all symbols have XKPHYS values and
the math fails for any other address range.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>


--- a/include/asm-mips/page.h	2007-09-26 05:51:10.000000000 +0000
+++ b/include/asm-mips/page.h	2007-09-26 05:51:24.000000000 +0000
@@ -149,7 +149,7 @@
 /*
  * __pa()/__va() should be used only during mem init.
  */
-#if defined(CONFIG_64BIT) && !defined(CONFIG_BUILD_ELF64)
+#ifdef CONFIG_64BIT
 #define __pa_page_offset(x)	((unsigned long)(x) < CKSEG0 ? PAGE_OFFSET : CKSEG0)
 #else
 #define __pa_page_offset(x)	PAGE_OFFSET

